diff --git a/node_modules/next/dist/esm/server/lib/incremental-cache/index.js b/node_modules/next/dist/esm/server/lib/incremental-cache/index.js
index e622976..588f542 100644
--- a/node_modules/next/dist/esm/server/lib/incremental-cache/index.js
+++ b/node_modules/next/dist/esm/server/lib/incremental-cache/index.js
@@ -349,9 +349,10 @@ export class IncrementalCache {
         }
         if (this.dev && !ctx.fetchCache) return;
         // fetchCache has upper limit of 2MB per-entry currently
-        if (ctx.fetchCache && JSON.stringify(data).length > 2 * 1024 * 1024) {
+        const max_fetch_size = Number(process.env.MAX_FETCH_SIZE) || 2 //gets value set in next.config
+        if (ctx.fetchCache && JSON.stringify(data).length > max_fetch_size * 1024 * 1024) {
             if (this.dev) {
-                throw new Error(`fetch for over 2MB of data can not be cached`);
+                throw new Error(`fetch for over ${max_fetch_size}MB of data can not be cached`);
             }
             return;
         }
diff --git a/node_modules/next/dist/server/lib/incremental-cache/index.js b/node_modules/next/dist/server/lib/incremental-cache/index.js
index 9f2935f..0308496 100644
--- a/node_modules/next/dist/server/lib/incremental-cache/index.js
+++ b/node_modules/next/dist/server/lib/incremental-cache/index.js
@@ -376,9 +376,10 @@ class IncrementalCache {
         }
         if (this.dev && !ctx.fetchCache) return;
         // fetchCache has upper limit of 2MB per-entry currently
-        if (ctx.fetchCache && JSON.stringify(data).length > 2 * 1024 * 1024) {
+        const max_fetch_size = Number(process.env.MAX_FETCH_SIZE) || 2 //gets value set in next.config
+        if (ctx.fetchCache && JSON.stringify(data).length > max_fetch_size * 1024 * 1024) {
             if (this.dev) {
-                throw new Error(`fetch for over 2MB of data can not be cached`);
+                throw new Error(`fetch for over ${max_fetch_size}MB of data can not be cached`);
             }
             return;
         }
